### PR TITLE
auth mydns: add SOA to list() output, fixes #9181

### LINKED ISSUE #9181: b2b-migrate mydns gmysql doesn't create SOA records
- Program: Authoritative
 - Issue type: Bug report

### Short description
When using pdnsutil to migrate backends from MyDNS to gmysql pdnsutil does not create an SOA record in the 'records' table. MyDNS uses the 'soa' table to define domains.

### Environment

 - Operating system: CentOS 8 
 - Software version: 4.2.2
 - Software source: PowerDNS repository

### Steps to reproduce

1. pdnsutil b2b-migrate mydns gmysql

### Expected behaviour
I would expect that an SOA record would be generated for each migrated domain.

### Actual behaviour
The domain record was created and then records were created but no SOA record was ever created.

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
Otherwise, the target database ends up with no SOA records for the migrated zones.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: modules/mydnsbackend/mydnsbackend.cc [hunk 1]
```diff
@@ -366,6 +373,24 @@ bool MyDNSBackend::get(DNSResourceRecord &rr) {
 
   SSqlStatement::row_t rrow;
 
+  if(!d_list_SOA_sent) {
```
Comment (jsoref) [inline modules/mydnsbackend/mydnsbackend.cc:376]:
```suggestion
  if (!d_list_SOA_sent) {
```

**Comment by zeha** [comment]
Travis shows:

```
[2020-06-02 20:33:39] any-nxdomain
[2020-06-02 20:33:39] --- ./tests/any-nxdomain/expected_result	2020-06-02 20:08:57.944775110 +0000
[2020-06-02 20:33:39] +++ ./tests/any-nxdomain/real_result	2020-06-02 20:33:37.495913686 +0000
[2020-06-02 20:33:39] @@ -1,4 +1,3 @@
[2020-06-02 20:33:39] -1	example.com.	IN	SOA	86400	ns1.example.com. ahu.example.com. 2847484148 28800 7200 604800 86400
[2020-06-02 20:33:39]  2	.	IN	OPT	32768	
[2020-06-02 20:33:39] -Rcode: 3 (Non-Existent domain), RD: 0, QR: 1, TC: 0, AA: 1, opcode: 0
[2020-06-02 20:33:39] +Rcode: 2 (Server Failure), RD: 0, QR: 1, TC: 0, AA: 1, opcode: 0
[2020-06-02 20:33:39]  Reply to question for qname='nxdomain.example.com.', qtype=ANY
[2020-06-02 20:33:40] ./start-test-stop: line 103:   311 Terminated              $RUNWRAPPER $PDNS --daemon=no --local-address=$address --local-port=$port --socket-dir=./ --config-dir=. --config-name=mydns --no-shuffle --master --cache-ttl=0 --dname-processing --mydns-rr-active=no --mydns-soa-active=no --mydns-use-minimal-ttl=no
[2020-06-02 20:33:41] ./start-test-stop: line 137: kill: (311) - No such process
```

File: modules/mydnsbackend/mydnsbackend.cc [hunk 2]
```diff
@@ -366,6 +372,25 @@ bool MyDNSBackend::get(DNSResourceRecord &rr) {
 
   SSqlStatement::row_t rrow;
 
+  if(d_send_SOA_first) {
```
Comment (jsoref) [inline modules/mydnsbackend/mydnsbackend.cc:375]:
```suggestion
  if (d_send_SOA_first) {
```

**Comment by rgacogne** [comment]
The failure on Travis seems to be a real issue:
```
[2020-06-03 07:41:45] any-nxdomain: 
[2020-06-03 07:41:45] Jun 03 07:41:45 Exception building answer packet for nxdomain.example.com/ANY (Attempt to print an unset dnsname) sending out servfail
[2020-06-03 07:41:45] Jun 03 07:41:45 Backend reported permanent error which prevented lookup (MyDNSBackend unable to lookup nxdomain.example.com: Attempt to bind more parameters than query has: SELECT type, data, aux, ttl, zone FROM `rr` WHERE zone = ? AND (name = ? OR name = ?) AND type = ? AND 1 = 1 ORDER BY type,aux,data), aborting
[2020-06-03 07:41:45] Jun 03 07:41:45 TCP nameserver had error, cycling backend: MyDNSBackend unable to lookup nxdomain.example.com: Attempt to bind more parameters than query has: SELECT type, data, aux, ttl, zone FROM `rr` WHERE zone = ? AND (name = ? OR name = ?) AND type = ? AND 1 = 1 ORDER BY type,aux,data
[2020-06-03 07:41:45] Jun 03 07:41:45 TCP server is without backend connections, launching
[2020-06-03 07:41:45] Jun 03 07:41:45 [MyDNSbackend] Connection successful
[2020-06-03 07:41:45] 	Failed test ./tests/any-nxdomain
```

**Comment by Habbie** [comment]
Yes :( something subtle, because it does work on CircleCI and my system. I'm debugging it on a separate branch.

**Pull request merged**